Intact.BuinessLogic/Services/ProfilesService.cs

```diff
@@ -0,0 +1,61 @@
+using Intact.BusinessLogic.Data;
+using Intact.BusinessLogic.Data.Enums;
+using Intact.BusinessLogic.Data.Models;
+using Intact.BusinessLogic.Mappers;
+using Intact.BusinessLogic.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace Intact.BusinessLogic.Services;
+
+public interface IProfilesService
+{
+    Task<IEnumerable<Profile>> GetAsync(Guid userId, CancellationToken cancellationToken);
+    
+    Task<Profile?> CreateAsync(Guid userId, string name, CancellationToken cancellationToken);
+    
+    Task DeleteAsync(Guid userId, string name, CancellationToken cancellationToken);
+}
+
+public class ProfilesService(AppDbContext appDbContext) : IProfilesService
+{
+    private const int PROFILES_MAX_COUNT_PER_USER = 5;
+    private readonly AppDbContext _appDbContext = appDbContext;
+
+    public async Task<IEnumerable<Profile>> GetAsync(Guid userId, CancellationToken cancellationToken)
+    {
+        return ProfileMapper.Map(await _appDbContext.ProfilesDaos
+            .Where(x => x.UserId == userId && x.State == ProfileState.Active)
+            .ToListAsync(cancellationToken: cancellationToken));
+    }
+
+    public async Task<Profile?> CreateAsync(Guid userId, string name, CancellationToken cancellationToken)
+    {
+        if ((await GetAsync(userId, cancellationToken)).Count() < PROFILES_MAX_COUNT_PER_USER)
```

Comment:
I think Count should contain condition to consider only active profiles, and not deleted, banned, etc.